plugins/woocommerce/includes/class-wc-install.php

```diff
@@ -347,8 +374,22 @@ public static function install() {
 
 		delete_transient( 'wc_installing' );
 
+		// Use add_option() here to avoid overwriting this value with each
+		// plugin version update. We base plugin age off of this value.
+		add_option( 'woocommerce_admin_install_timestamp', time() );
+
 		do_action( 'woocommerce_flush_rewrite_rules' );
 		do_action( 'woocommerce_installed' );
+		do_action( 'woocommerce_admin_installed' );
+	}
+
+	/**
+	 * Returns true if we're installing.
+	 *
+	 * @return bool
+	 */
+	public static function is_installing() {
```

Comment:
I know this is copied from Woo Admin, but I think making this `public` instead of `private` encourages a bad pattern in other classes to check if WooCommerce is installing.

Instead I think other classes instances should be created when it is safe, from the `woocommerce_loaded` hook for example, which would avoid the need for the class to concern itself with whether WooCommerce is installing.

It looks like `DeprecatedOptions` is using this pattern. Possibly, we could just have it checking `get_transient( 'wc_installing' )` for now and follow-up on improving it later.

Reply:
Thank you for the suggestion 👍 

Updated in https://github.com/woocommerce/woocommerce/pull/32513/commits/818a1dcff78bf63e92ddc13d82b808538b5d7748

I think we can use `WC_INSTALLING` constant instead of checking the transient value.